### DISCUSSION TITLE
node hangs while trying to create a draft email

### DISCUSSION THREAD

**Discussion by aalmans** [umanamente/n8n-nodes-imap#37] [General]
Hi there,

I want to use this module to put a draft email in my IMAP server so I can review it and send it later. I am using the RFC822 option to send plaintext email.

When I put a fixed value in the body text the imap node sends it ok.

But when I populate the body with a value from a previous node it hangs indefinitely. 

Any idea why?

**Comment by umanamente** [comment]
Could you send a screenshot of your IMAP node's parameters?

**Comment by aalmans** [comment]
Sure, here it is. I removed the sender email from the image though.

![image](https://github.com/user-attachments/assets/70cc8694-62c3-4620-81a2-1d3c2c04a54e)

When I replace the variable with static text it sends just fine. When I leave the variable in there, the node keeps spinning forever.

**Reply by umanamente** [reply]
Please add an additional "Set" node before the IMAP node and compose the content into a variable. Use this variable later in the IMAP node.
Once done, send the generated content here and check if it differs from the static content.

**Comment by aalmans** [comment]
Here is the output. It is still spinning:

![image](https://github.com/user-attachments/assets/ec9c9990-7a3d-4280-be45-0171f754a46d)

**Reply by umanamente** [reply]
If you insert the same text into the field without {{ }}, does the IMAP node work properly?

**Comment by aalmans** [comment]
No,

When i paste in the text:

Hi \n\nI hope you’re doing well. I wanted to reach out one last time about a possible collaboration. I think it would be great for both of our sites.

it also spins. Also when I remove the \n's

However, when I type the same text in that window, so not paste in the text, it sends ok

**Comment by aalmans** [comment]
Hi there, i was wondering if there was an update on this maybe? I am currently using n8n to send to make.com to use an imap node there to create the draft but i'd rather use your module and stay within n8n

**Reply by umanamente** [reply]
The custom node works as expected when using plain text, but fails when the same text includes variable substitutions with double curly braces. This suggests the issue is likely related to your N8N instance configuration.

Please reach out to the N8N support team on [their forum](https://community.n8n.io/c/questions/12) to see if they can provide any insights on this problem. Once you’ve created the forum post, please share the link here so I can follow the discussion.
